internal/util/paths/paths.go

```diff
@@ -79,17 +79,30 @@ func WriteString(path string, content string) error {
 	return err
 }
 
-// Expands leading tilde.
-func ExpandTilde(path string) (string, error) {
-	usr, err := user.Current()
-	if err != nil {
-		return "", fmt.Errorf("failed to determine current user")
-	}
-	home := usr.HomeDir
-	if path == "~" {
-		path = home
-	} else if strings.HasPrefix(path, "~/") {
-		path = filepath.Join(home, path[2:])
+// Expands environment variables and `~`, returning an absolute path.
+// On error, an empty string is returned.
+func ExpandPath(path string) (string, error) {
+
+	if strings.HasPrefix(path, "~/") {
+		// resolve necessary variables
+		usr, err := user.Current()
+		if err != nil {
+			return "", err
+		}
+		home := usr.HomeDir
+		if home == "" {
+			return "", errors.New("Could not find user's home directory.")
+		}
+
+		// construct as abs path
+		if path == "~" {
```

Comment:
This condition is always false, because this is inside `if strings.HasPrefix(path, "~/") {`.

You probably want: `if strings.HasPrefix(path, "~") {`. above.

Reply:
ooh yes 👌